app/controllers/comments_controller.rb

```diff
@@ -1,13 +1,24 @@
 class CommentsController < ApplicationController
   load_and_authorize_resource
   before_action :set_user_and_post
+
+  def index
+    @comment = Comment.new
+    render json: @comment, only: [:Text]
+  end
+
   def new
     @comment = Comment.new
   end
 
   def create
-    @comment = @post.comments.build(params.require(:comment).permit(:Text).merge(user_id: current_user.id))
+    @comment = @post.comments.build(comment_params.merge(user_id: current_user.id))
     if @comment.save
+
+      respond_to do |format|
+        format.html { user_post_path(@post, @user) }
+        format.json { render json: @comment }
+      end
       flash[:success] = 'Comment saved successfully'
       redirect_to user_post_path(@post, @user)
```

Comment:
- [ ] I took the pleasure of trying to create a comment for a `Post` instance even after skipping the **_CRSF token_** check and I got this error most times:

![image](https://github.com/ismayito/Rails-blog-app/assets/77106962/00d16f80-9207-4c10-807a-b964d47e1fef)

--- 
Remember that you have to explicitly state the `User`, instead of saying the `current_user.id`. Therefore to fix this, I will advise you to do the following 👍 💻:

```ruby
 @comment = @post.comments.build(comment_params.merge(user_id: @user.id))
```